### PR TITLE
Upgrade UglifyJS to 3.10.0, fixing compatibility with that version

### LINKED ISSUE #14779: Upgrade Uglify-js to 3.10.0 or higher and fix our assets tests coverage
Upgrading to uglify-js 3.10.0 caused https://github.com/mozilla/addons-server/issues/14726. It didn't break any tests.
We are reverting this upgrade, but then we should:
- Figure out why this didn't work and fix the issue
- Figure out why this didn't break any tests and fix the tests

### PR DESCRIPTION
**Pull request by diox**
Also added a test proving we do compress js files correctly.

Fixes #14779

### CODE REVIEW

File: src/olympia/amo/management/commands/compress_assets.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def _minify(self, ftype, file_in, file_out):
         """Run the proper minifier on the file."""
         if ftype == 'js' and hasattr(settings, 'UGLIFY_BIN'):
             opts = {'method': 'UglifyJS', 'bin': settings.UGLIFY_BIN}
-            run_command('{uglify} -v -o {target} {source} -m'.format(
+            run_command('{uglify} -o {target} {source} -m'.format(
```
Comment (bobsilverberg) [inline src/olympia/amo/management/commands/compress_assets.py:201]: You decided not to bother with `--verbose`?

Comment (diox) [reply src/olympia/amo/management/commands/compress_assets.py:201]: I tried it and there wasn't any difference in the output, weirdly. I decided I had already spent too much time on this...

File: src/olympia/amo/tests/test_commands.py [hunk 2]
```diff
@@ -115,6 +125,43 @@ def test_compress_assets_correctly_fetches_static_images(settings, tmpdir):
...
+    except FileNotFoundError:
+        pass
+
+    # Capture output to avoid it being logged and allow us to validate it
+    # later if needed
+    out = io.StringIO()
+
+    # Now run compress and collectstatic
+    call_command('compress_assets', force=True, stdout=out)
+    call_command('collectstatic', interactive=False, stdout=out)
```
Comment (bobsilverberg) [inline src/olympia/amo/tests/test_commands.py:158]: I assume we need to call both of these to get the files to be written?

Comment (diox) [reply src/olympia/amo/tests/test_commands.py:158]: Technically, only the first one is needed to have files written, but `collecstatic` moves static files from all apps to the final location where they will be served from, so it's useful to have this here as well to test the full config/behavior.

**Pull request merged**